client/driver/docker.go

```diff
@@ -168,9 +168,11 @@ type DockerLoggingOpts struct {
 }
 
 type DockerMount struct {
+	Type          string                 `mapstructure:"type"`
```

Comment:
`type` should imo be defaulted to `volume` to keep it compat for now

Reply:
Agree.  Added in https://github.com/hashicorp/nomad/pull/4761/commits/4432486f490309dbcc43c9cc8ac0faae5af55d9b#diff-a6757d41ad7cd7f37f0f0243e3575307R403